common/scala/src/main/scala/org/apache/openwhisk/core/connector/Message.scala

```diff
@@ -428,6 +428,61 @@ object EventMessage extends DefaultJsonProtocol {
   def parse(msg: String) = Try(format.read(msg.parseJson))
 }
 
+case class InvokerResourceMessage(status: String,
+                                  freeMemory: Long,
+                                  busyMemory: Long,
+                                  inProgressMemory: Long,
+                                  tags: Seq[String],
+                                  dedicatedNamespaces: Seq[String])
+    extends Message {
+
+  /**
+   * Serializes message to string. Must be idempotent.
+   */
+  override def serialize: String = InvokerResourceMessage.serdes.write(this).compactPrint
+
+  def canEqual(a: Any) = a.isInstanceOf[InvokerResourceMessage]
+
+  override def equals(that: Any): Boolean =
```

Comment:
is this method needed? cases class equals already provides this implementation by definition.

Reply:
Ah, yes, i tested in my local, after removed `the override def equals and the override def hashCode`, works well both, e.g.
for two InvokerResourceMessage object,
* If all field value is same, object1.equals(object) will be `true`, 
* if any one filed value is different,  object1.equals(object) will be `false`